apps/website/src/pages/clips/index.tsx

```diff
@@ -0,0 +1,330 @@
+  clip: Clip;
+  clipClicked: (clip: Clip) => void;
+  lastItemRef: React.RefObject<HTMLDivElement>;
+}) => {
+  const { data: session } = useSession();
+
+  const addVote = trpc.clips.addVote.useMutation({
+    onSuccess: () => {
+      setHasVoted(true);
+      clip.voteCount++;
+    },
+  });
+
+  const removeVote = trpc.clips.removeVote.useMutation({
+    onSuccess: () => {
+      setHasVoted(false);
+      clip.voteCount--;
+    },
+  });
+
+  const handleVote = (clipId: string) => {
+    if (hasVoted) {
+      removeVote.mutate({ clipId });
+    } else {
+      addVote.mutate({ clipId });
+    }
+  };
+
+  const [hasVoted, setHasVoted] = useState(clip.hasVoted);
+  const thumbsClass = hasVoted
+    ? "hover:fill-alveus-green-900 fill-green"
+    : "hover:fill-green fill:alveus-green-900";
+
+  return (
+    <div>
+      <div
+        ref={lastItemRef}
+        className="group cursor-pointer"
+        onClick={() => clipClicked(clip)}
+      >
+        <Image
+          src={clip.thumbnailUrl}
+          alt=""
+          width={700}
+          height={300}
+          className="h-auto w-full rounded-xl transition group-hover:scale-102 group-hover:shadow-lg"
+        />
+        <div className="min-h-[4rem]">
+          <p className="text-center text-xl font-medium transition-colors group-hover:text-alveus-green-700 ">
+            {clip.title}
+          </p>
+        </div>
+      </div>
+      <div className="flex justify-between px-4">
+        <p className="text-left">Votes: {clip.voteCount}</p>
+        {session && (
+          <p
+            className={classes("cursor-pointer text-right", thumbsClass)}
+            onClick={() => handleVote(clip.id)}
+          >
+            <IconThumbsUp size={20} />
+          </p>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const ClipModal = ({
+  clip,
+  onClose,
+}: {
+  clip: Clip;
+  onClose: (e: React.MouseEvent) => void;
+}) => {
+  const clipUrl = new URL("https://clips.twitch.tv/embed");
+  clipUrl.searchParams.set("clip", clip.slug);
+  clipUrl.searchParams.set("parent", window.location.hostname);
+
+  return (
+    <div
+      className="fixed inset-0 z-50 flex overflow-auto bg-black/75"
+      onClickCapture={onClose}
+    >
+      <div className="relative m-auto flex max-h-[50vh] w-full flex-col rounded-lg border border-solid border-black bg-white px-2 pb-4 pt-10 sm:px-10 sm:py-10 md:max-w-[75vw] lg:max-h-[75vh] lg:max-w-[60vw] 2xl:max-w-[50vw]">
+        <span className="absolute right-0 top-0 p-2" onClick={onClose}>
+          <svg
+            onClick={onClose}
+            className="h-6 w-6 cursor-pointer text-gray-700"
+            fill="none"
+            viewBox="0 0 24 24"
+            stroke="currentColor"
+          >
+            <path
+              onClick={onClose}
+              strokeLinecap="round"
+              strokeLinejoin="round"
+              strokeWidth="2"
+              d="M6 18L18 6M6 6l12 12"
+            />
+          </svg>
+        </span>
```

Comment:
dito use a `<button>`. the events should bubble up, attaching the events to the svg and path should not be necessary